libqtile/widget/redshift.py

```diff
@@ -0,0 +1,333 @@
+                "redshift: format_object: obj didn't match any of the instance types, returning empty string"
+            )
+            return ""
+
+    @expose_command
+    def scroll_up(self):
+        """
+        Scroll up to next item.
+        """
+        self._scroll(1)
+
+    @expose_command
+    def scroll_down(self):
+        """
+        Scroll down to next item.
+        """
+        self._scroll(-1)
+
+    def show_line(self):
+        """
+        Update the text with the the current index in lines.
+        """
+        if not self._lines:
+            return
+
+        obj = self._lines[self._line_index]
+
+        self.update(self.format_object(obj))
+
+    @expose_command
+    def enable_redshift(self):
+        """
+        Enable redshift with the temp, gamma and brightness params.
+        """
+        gamma_val = GammaGroup(self.gamma_red, self.gamma_green, self.gamma_blue)
+        try:
+            self._get_rs_driver().enable(self.temperature, gamma_val, self.brightness)
+        except Exception as e:
+            raise RuntimeError(f"redshift: could not enable redshift: {e}")
+
+    @expose_command
+    def click(self):
+        """
+        Either enable or disable the widget ONLY if we are currently on the
+        first line ie. the enabled/disabled text.
+        """
+        if self._line_index != 0:
+            return
+        if self.is_enabled:
+            self.reset_redshift()
+        else:
+            self.enable_redshift()
+        self.is_enabled = not self.is_enabled
+        self._set_text()
+        self.bar.draw()
+
+    @expose_command
+    def reset_redshift(self):
+        """
+        Make a call to redshift to reset everything.
+        """
+        try:
+            self._get_rs_driver().reset()
+        except Exception as e:
+            raise RuntimeError(f"redshift: could not reset redshift: {e}")
+
+    def _scroll(self, step):
+        """
+        Scroll up/down dictated by 'step' the items
+        and then display that item.
+        """
+        # Make sure that the lines var is set, maybe
+        # some of the values have been updated?
+        self._set_lines()
+        self._line_index = (self._line_index + step) % len(self._lines)
+        self.show_line()
+
+    def _set_lines(self):
+        """
+        Set the list of formatted text items to scroll through.
+        """
+        first_text = ""
+        if self.is_enabled:
+            first_text = self.enabled_txt
+        else:
+            first_text = self.disabled_txt
+
+        first_text = self._format_first_text(str(first_text))
+
+        self._lines = [
+            first_text,
+            self._format_brightness_text(),
+            self._format_temp_text(),
+            self._format_gamma_text(
+                GammaGroup(self.gamma_red, self.gamma_green, self.gamma_blue)
+            ),
+        ]
+
+    def _set_text(self):
+        """
+        Set text (not update) to the current index of lines.
```

Comment:
Why `not update`? Is this because you call the line during `_configure` and so don't want to trigger a draw of the bar? That's easy enough to fix - see my suggestion in `_configure`.

Reply:
Thanks, done